core/src/main/scala/org/apache/spark/scheduler/LiveListenerBus.scala

```diff
@@ -124,11 +136,13 @@ private[spark] class LiveListenerBus(val sparkContext: SparkContext) extends Spa
       logError(s"$name has already stopped! Dropping event $event")
       return
     }
+    metrics.numEventsReceived.inc()
```

Comment:
Yes. My idea was to have a counter which is incremented whenever an event is received, regardless of how it ends up being processed.